float8_experimental/fsdp_utils.py

```diff
@@ -0,0 +1,61 @@
+import math
+import warnings
+from typing import List
+
+import torch
+import torch.nn as nn
+from float8_experimental.float8_dynamic_utils import WeightWithDynamicFloat8CastTensor
+from float8_experimental.float8_linear import Float8Linear
+from float8_experimental.float8_linear_utils import linear_requires_sync
+from float8_experimental.float8_utils import EPS
+
+
+def precompute_float8_scale_for_fsdp(module: nn.Module) -> None:
+    """
+    Calculate scale for all float8 parameters after optimizer step
+    It performs a single all-reduce instead of many all-reduces for each parameter
+    Exmaple usage:
+        model(input).sum().backward()
+        optim.step()
+        precompute_float8_scale_for_fsdp(model)
+    """
+    from torch.distributed._tensor import DTensor
+
+    if any(
```

Comment:
I think this can be relaxed to something like

```
if isinstance(m, Float8Linear) and m.scaling_type_w is TensorScalingType.DELAYED:
    raise NotImplementedError(...)
```

the scaling of activations and gradient's should not matter for this case